src/main/scala/io/viash/config/dependencies/AbstractGitRepository.scala

```diff
@@ -59,22 +66,75 @@ trait AbstractGitRepository extends Repository with Logging {
       case Some(value) => List("--single-branch", "--branch", value)
     }
 
-    val loggers = Seq[String => Unit] { (str: String) => {info(str)} }
     Exec.runCatch(
       List("git", "clone", uri, "--no-checkout", "--depth", "1") ++ singleBranch :+ ".",
       cwd = cwd,
-      loggers = loggers,
+      loggers = getLoggers("doGitClone"),
     )
   }
 
   protected def checkGitAuthentication(uri: String): Boolean = {
     val res = Exec.runCatch(
       List("git", "ls-remote", uri),
+      loggers = getLoggers("checkGitAuthentication"),
     )
     res.exitValue == 0
   }
 
+  protected def getLocalHash(): String = {
+    val cwd = Some(Paths.get(localPath).toFile())
+    val res = Exec.runCatch(
+      List("git", "rev-list", "-n", "1", "HEAD"),
+      cwd = cwd,
+      loggers = getLoggers("getLocalHash"),
+    )
+    res.exitValue match {
+      case 0 => res.output.trim()
+      case _ => "Local hash not found"
+    }
+  }
+
+  protected def getRemoteHash(uri: String): String = {
+    val res = Exec.runCatch(
+      List("git", "ls-remote", uri, tag.getOrElse("HEAD")),
+      loggers = getLoggers("getRemoteHash"),
+    )
+    res.exitValue match {
+      case 0 => res.output.split("\t").head
+      case _ => "Remote hash not found"
+    }
+  }
+
   def getCheckoutUri(): String
+  def getCacheIdentifier(): Option[String]
+  def fullCachePath: Option[Path] = {
+    val cacheIdentifier = getCacheIdentifier()
+    cacheIdentifier.map(cacheIdentifier => Paths.get(SysEnv.viashHome).resolve("cache").resolve(cacheIdentifier))
```

Comment:
I'd be inclined to propose to cache them to `~/.viash/repositories/git-...`, or `repository_cache`.

```suggestion
    cacheIdentifier.map(cacheIdentifier => Paths.get(SysEnv.viashHome).resolve("repositories").resolve(cacheIdentifier))
```

Reply:
tomato potato ;) ... but sure!